core/src/main/scala/kafka/log/LogManager.scala

```diff
@@ -503,14 +642,14 @@ class LogManager(val logDirs: Array[File],
    * data directory with the fewest partitions.
    */
   private def nextLogDir(): File = {
-    if(logDirs.size == 1) {
-      logDirs(0)
+    if(_liveLogDirs.size == 1) {
```

Comment:
Yeah I think it is useful. It is probably better to do it in KIP-113 so that we can finish this KIP sooner :)